cip/1.accepted/CIP2017-03-29-Single-Value-Subqueries.adoc

```diff
@@ -0,0 +1,68 @@
+= CIP2017-03-29 Scalar Subqueries and List Subqueries
+:numbered:
+:toc:
+:toc-placement: macro
+:source-highlighter: codemirror
+
+*Author:* Tobias Lindaaker <tobias.lindaaker@neotechnology.com>
+
+toc::[]
+
+== Scalar Subqueries and List Subqueries
+
+Scalar Subqueries are read-only subqueries that produce a single value in a single row.
+The result of a Scalar Subquery is the single value (in the single row) produced by the subquery.
+
+List Subqueries are read-only subqueries that produce a single value per row, and zero or more rows.
+The result of a List Subquery is to collect the value of all rows produced by the subquery into a list.
+
+=== Syntax
+
+[source, ebnf]
+----
+ListSubquery = '[', SingleValueSubquery, ']' ;
+
+ScalarSubquery = '<(((<', SingleValueSubquery, '>)))>' ;
```

Comment:
One more... if we don't allow subquery short forms, i.e. starting with a pattern, round parenthesis might actually suffice from a parsing point of view... `(MATCH (a)-[r]->(b) RETURN r)`. It's inconsistent with the other subquery forms under discussion but it would save us the hassle of coming up with weird fancy bracket syntax.